### PR TITLE
Upgrade `colony-js` to v5

### LINKED ISSUE #7: Use v5 branch of ColonyJS
The only thing to pay attention here is that the `ColonyManager` lib will still work after this

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Updated `@colony/colony-js` to v5, along with `node-js` to 16.14 as it's required by `colony-js` v5

Also:
- `ROOT_DOMAIN_ID` was replaced by `Id.RootDomain`.
- The `TransactionOverrides` type is no longer available via `colony-js`, now we are going to use the `Overrides` type made available by `ethers` instead.
- `ColonyClient` type was replaced by `AnyColonyClient`.
- `ColonyVersion` no longer associates version numbers with version names.
- Now `AnyVotingReputationClient` type is available, it works the same way as `AnyColonyClient`.
- `getMoveFundsPermissionProofs` is no longer available, the only place where we used it was `moveFundsMotion` and as a replacement for it now we are going to use `moveFundsBetweenPotsWithProofs`.
- RIP `ContractClient`, now it's replaced by the `Contract` type available on `ethers`.
- `Network.XdaiFork` was replaced by `Network.XdaiQa`.
- `colonyNetworkAddresses` was replaced by `ColonyNetworkAddress`
- `Network.Local` was replaced by `Network.Custom`.

Resolves #7

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 A interation of the Colony Dapp sporting both a fully decentralized operating mode, as well as a mode enhanced by a metadata caching layer
 
 ## Prerequisites
-* `node` `v14.18.x` (Best use [nvm](https://github.com/nvm-sh/nvm))
+* `node` `v16.14.x` (Best use [nvm](https://github.com/nvm-sh/nvm))
```
Comment (chinins) [inline Readme.md:6]: Great, that you remembered about updating the readme

File: src/redux/sagas/motions/moveFundsMotion.ts [hunk 2]
```diff
@@ -122,22 +103,12 @@ function* moveFundsMotion({
...
-        fromChildSkillIndex,
-        toChildSkillIndex,
-        fromPot,
-        toPot,
-        amount,
-        tokenAddress,
-      ],
+        ? `moveFundsBetweenPotsWithProofs(uint256,uint256,uint256,address)`
+        : 'moveFundsBetweenPotsWithProofs',
+      [fromPot, toPot, amount, tokenAddress],
```
Comment (chinins) [inline src/redux/sagas/motions/moveFundsMotion.ts:111]: Lighter version - I like it. Hope it will work properly.

**Review by chinins — APPROVED** [review]
Very nice!

**Review by rdig — APPROVED** [review]
Looks good.

However I do suspect that we're going to have to come back to this once the CDapp is up and running, as I forsee us needing to make some changes to `colonyJS` to accomodate some of our use cases.

File: Readme.md (hunk 1 above)
Comment (ArmandoGraterol) [reply Readme.md:6]: Yeah I almost forgot to do it :P

**Pull request merged**